crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -243,6 +252,48 @@ where
     fn key(key_index: u16) -> Vec<u8> {
         (Self::KEY_PREFIX, key_index.to_le_bytes()).encode()
     }
+
+    /// Get segment headers that are expected to be included at specified block number.
+    pub fn segment_headers_for_block(&self, block_number: BlockNumber) -> Vec<SegmentHeader> {
+        let Some(last_segment_index) = self.max_segment_index() else {
+            return Vec::new(); // not initialized
+        };
+
+        // Special case for the initial segment (for genesis block).
+        if block_number == 1 {
+            return vec![self
+                .get_segment_header(SegmentIndex::ZERO)
+                .expect("The initial segment header must exist at this point.")];
+        }
+
+        let mut current_segment_index = last_segment_index;
+        loop {
+            let current_segment_header = self
+                .get_segment_header(current_segment_index)
+                .expect("Current segment must always exist.");
+
+            // The block immediately after the archived segment adding the confirmation depth
+            let target_block_number =
+                current_segment_header.last_archived_block().number + 1 + self.confirmation_depth_k;
+            if target_block_number == block_number {
+                return vec![current_segment_header];
```

Comment:
I'd like to see a change that supports a hypothetical case of block exceeding segment header size here, resulting in more than one segment header returned.